src/protocols/LinuxDMABUF.cpp

```diff
@@ -290,6 +288,8 @@ CLinuxDMABUFFeedbackResource::CLinuxDMABUFFeedbackResource(SP<CZwpLinuxDmabufFee
         .data = (void*)&feedback->mainDevice,
     };
     resource->sendMainDevice(&deviceArr);
+
+    // Main tranche
     resource->sendTrancheTargetDevice(&deviceArr);
     resource->sendTrancheFlags((zwpLinuxDmabufFeedbackV1TrancheFlags)0);
```

Comment:
line 294 is undefined behaviour btw, `(zwpLinuxDmabufFeedbackV1TrancheFlags)0`  casting enum out of range.

```
enum zwpLinuxDmabufFeedbackV1TrancheFlags : uint32_t {
    ZWP_LINUX_DMABUF_FEEDBACK_V1_TRANCHE_FLAGS_SCANOUT = 1,
};
```

Reply:
what am I supposed to do if I want to send no flags then?

Reply:
just checking weston a bit should that even be a thing? https://gitlab.freedesktop.org/wayland/weston/-/blob/main/libweston/linux-dmabuf.c#L444 seems like the tranche doesnt get created if there is no flags, and ofc later sends it https://gitlab.freedesktop.org/wayland/weston/-/blob/main/libweston/linux-dmabuf.c#L770 if im not misreading this

Reply:
or if really needed either make sendtrancheflags use uint directly or add a NONE = 0 to the enum

Reply:
I think you just don't call `resource->sendTrancheFlags` if you don't wanna hint for `scanout`. I see that kwin never does it 
https://github.com/KDE/kwin/blob/50717a2f9ac79b8131ad2220003ed971d74f01d7/src/wayland/linuxdmabufv1clientbuffer.cpp#L204

Reply:
makes sense

Reply:
kwin also seems to be doing something like this regarding sending formats and modifiers

```
diff --git a/src/protocols/LinuxDMABUF.cpp b/src/protocols/LinuxDMABUF.cpp
index 994882c81..6ab9cd6e1 100644
--- a/src/protocols/LinuxDMABUF.cpp
+++ b/src/protocols/LinuxDMABUF.cpp
@@ -381,14 +381,12 @@ bool CLinuxDMABUFResource::good() {
 void CLinuxDMABUFResource::sendMods() {
     for (auto& [fmt, mod] : PROTO::linuxDma->defaultFeedback->tranches) {
         if (resource->version() < 3) {
-            if (mod == DRM_FORMAT_MOD_INVALID)
+            if (mod == DRM_FORMAT_MOD_INVALID || mod == DRM_FORMAT_MOD_LINEAR)
                 resource->sendFormat(fmt);
-            continue;
+        } else {
+            // TODO: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1166
+            resource->sendModifier(fmt, mod >> 32, mod & 0xFFFFFFFF);
         }
-
-        // TODO: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1166
-
-        resource->sendModifier(fmt, mod >> 32, mod & 0xFFFFFFFF);
     }
 }
 ```
 
 only send formats if version < 3, and also send formats on DRM_FORMAT_MOD_LINEAR

Reply:
pushed